### PR TITLE
Fix(dropdown): Use HostBinding and toggle event instead of renderer

### PR DESCRIPTION
**Pull request by Hypercubed**

### CODE REVIEW

**Comment by amcdnl** [comment]
This is done for performance. The body listener only listens when it's open.

**Comment by Hypercubed** [comment]
@amcdnl I understand, but it does not work in some instances.

**Comment by amcdnl** [comment]
I strongly urge against a listener on the body that is always on.

**Comment by Hypercubed** [comment]
The existing implementation, in some cases, fails to trigger change detection after the call back.  An alternative is to add `this.cd.markForCheck` to each callback.

**Comment by amcdnl** [comment]
I think that would be much better solution.

File: src/components/dropdown/dropdown.component.ts [hunk 1]
```diff
@@ -46,33 +46,42 @@ export class DropdownComponent implements AfterContentInit, OnDestroy {
...
       }
     }
   }
 
   onToggleClick(ev): void {
     if(!this.dropdownToggle.disabled) {
       this.open = !this.open;
+
+      if (this.open) {
+        this.documentListener = this.renderer.listen(
```
Comment (amcdnl) [inline src/components/dropdown/dropdown.component.ts]: You should look at using RX for listening to the stream of click events.

Comment (Hypercubed) [reply src/components/dropdown/dropdown.component.ts]: I'll look into it... should be done across components.

**Pull request merged**